frontend/templates/product-list/ProductListView.tsx

```diff
@@ -21,24 +21,28 @@ import {
    ProductListChildrenSection,
    RelatedPostsSection,
 } from './sections';
+import { useVariantProductList } from './hooks/useVariantProductList';
 
 const HITS_PER_PAGE = 24;
 
 export interface ProductListViewProps {
    productList: ProductList;
    algoliaSSR?: boolean;
+   deviceHandle?: string;
 }
 
 export function ProductListView({
    productList,
    algoliaSSR,
+   deviceHandle,
 }: ProductListViewProps) {
    // This temporary hack allows to correctly populate the itemType facet during SSR
    // see: https://github.com/algolia/instantsearch/issues/5571
    const _ = useMenu({ attribute: 'facet_tags.Item Type' });
    const filters = computeProductListAlgoliaFilterPreset(productList);
    const optionalFilters =
       computeProductListAlgoliaOptionalFilters(productList);
+   productList = useVariantProductList(productList, deviceHandle);
    const itemTypeProductList = useItemTypeProductList(productList);
```

Comment:
I think we should have a single hook that augments the product list. As implemented, calling these hooks out of order would introduce a bug ("Device Item Type Variant" versus "Device Variant Item Type").

Reply:
It would also make the code easier to understand if all the logic that changes the `h1`, `title`, etc was side by side in a single place.

Reply:
I like the separation where one function that handles variant overrides and one that handles item type overrides, personally. But yeah i should make it more clear the order matters.